### PR TITLE
add option to specify input dir

### PR DESCRIPTION
**Pull request by keertip**

### CODE REVIEW

File: bin/dartdoc.dart [hunk 1]
```diff
@@ -44,6 +44,16 @@ void main(List<String> arguments) {
     exit(1);
   }
 
+  var inputDir;
```
Comment (sethladd) [inline bin/dartdoc.dart]: let's use annotations here. I've learned my lesson when we omit them :)

File: bin/dartdoc.dart [hunk 2]
```diff
@@ -44,6 +44,16 @@ void main(List<String> arguments) {
...
+  var input_dir = args['input-dir'];
```
Comment (sethladd) [inline bin/dartdoc.dart]: what's the difference between this one and inputDir? Are there more descriptive variable names?

File: bin/dartdoc.dart [hunk 3]
```diff
@@ -76,22 +88,24 @@ void _printUsageAndExit(ArgParser parser) {
...
   parser.addFlag('version',
       help: 'Display the version for $NAME.', negatable: false);
+  parser.addOption('dart-sdk',
+      help: "Location of the Dart SDK. Use if SDK isn't automatically located.");
   parser.addFlag('sdk-docs', help: 'Generate ONLY the docs for the Dart SDK.');
   parser.addOption('sdk-readme',
       help: 'Path to the SDK description file. Use if generating Dart SDK docs.');
-  parser.addOption('footer',
-      help: 'Path to an HTML, inserted into the footer of every page.');
+  parser.addOption('input-dir',
```
Comment (sethladd) [inline bin/dartdoc.dart]: if we're going to use `input-dir`, let's also use `output-dir`. I think `dir` is extraneous, so my vote is for `input`. Thoughts?

File: bin/dartdoc.dart [hunk 4]
```diff
@@ -76,22 +88,24 @@ void _printUsageAndExit(ArgParser parser) {
...
+      help: 'Path to source directory, defaults to current directory');
   parser.addOption('output',
       help: 'Path to output directory.', defaultsTo: 'docs');
+  parser.addOption('footer',
+      help: 'HTML text, inserted into the footer of every page.');
```
Comment (sethladd) [inline bin/dartdoc.dart]: Oh, not a path to a file? This might be hard to pass into the command-line, as our uses will want to pass in multi-line footers

Comment (keertip) [inline bin/dartdoc.dart]: has to be changed to path to file. Will do that when adding the option for header.

File: bin/dartdoc.dart (hunk 3 above)
Comment (keertip) [inline bin/dartdoc.dart]: sure, will change

File: bin/dartdoc.dart (hunk 4 above)
Comment (sethladd) [inline bin/dartdoc.dart]: Oh, maybe I was confused. What do I actually pass in on the command line? A path or the actual contents?

File: bin/dartdoc.dart (hunk 1 above)
Comment (keertip) [inline bin/dartdoc.dart]: done!

File: bin/dartdoc.dart (hunk 2 above)
Comment (keertip) [inline bin/dartdoc.dart]: cleaned up

File: bin/dartdoc.dart (hunk 4 above)
Comment (keertip) [inline bin/dartdoc.dart]: Sorry, I was not clear. Right now you have to pass the HTML text in on command line. This is not ideal, and will be changed to pass in path to a file with the contents.

File: bin/dartdoc.dart [hunk 5]
```diff
@@ -76,22 +88,24 @@ void _printUsageAndExit(ArgParser parser) {
...
       help: 'Display the version for $NAME.', negatable: false);
+  parser.addOption('dart-sdk',
+      help: "Location of the Dart SDK. Use if SDK isn't automatically located.");
   parser.addFlag('sdk-docs', help: 'Generate ONLY the docs for the Dart SDK.');
   parser.addOption('sdk-readme',
       help: 'Path to the SDK description file. Use if generating Dart SDK docs.');
-  parser.addOption('footer',
-      help: 'Path to an HTML, inserted into the footer of every page.');
+  parser.addOption('input',
+      help: 'Path to source directory, defaults to current directory');
```
Comment (sethladd) [inline bin/dartdoc.dart]: should we use the `defaultsTo` option?

File: bin/dartdoc.dart (hunk 4 above)
Comment (sethladd) [inline bin/dartdoc.dart]: Gotcha. Thanks! :)

File: bin/dartdoc.dart (hunk 5 above)
Comment (keertip) [inline bin/dartdoc.dart]: yes, thanks!

**Comment by sethladd** [comment]
lgtm!

**Pull request merged**